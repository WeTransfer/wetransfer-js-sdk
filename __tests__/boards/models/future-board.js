const { futureBoard } = require('../../../src/boards/models');

describe('Future board normalizer', () => {
  describe('normalizeBoard function', () => {
    let board = {};
    beforeEach(() => {
      board = {
        name: 'WeTransfer rocks',
        description: '',
      };
    });

    it('should return a normalized board', () => {
      const normalized = futureBoard(board);
      expect(normalized).toMatchSnapshot();
    });

    it('should remove extra properties', () => {
      const extraProps = Object.assign({}, board, { date: new Date() });
      const normalized = futureBoard(extraProps);
      expect(normalized).toMatchSnapshot();
    });
  });
});
