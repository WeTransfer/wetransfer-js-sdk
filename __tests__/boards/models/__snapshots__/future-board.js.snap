// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Future board normalizer normalizeBoard function should remove extra properties 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;

exports[`Future board normalizer normalizeBoard function should return a normalized board 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;
